coreneuron/io/mech_report.cpp

```diff
@@ -56,14 +56,16 @@ void write_mech_report() {
 
     /// print global stats to stdout
     if (nrnmpi_myid == 0) {
-        printf("\n================= MECHANISMS COUNT BY TYPE ===================\n");
+        printf("\n============== MECHANISMS COUNT AND SIZE BY TYPE =============\n");
         printf("%4s %20s %10s %25s\n", "Id", "Name", "Count", "Total memory size (KiB)");
         for (size_t i = 0; i < total_mech_count.size(); i++) {
-            printf("%4lu %20s %10ld %25.2lf\n",
-                   i,
-                   nrn_get_mechname(i),
-                   total_mech_count[i],
-                   static_cast<double>(total_mech_size[i]) / 1024);
+            if (total_mech_count[i] > 0) {
+                printf("%4lu %20s %10ld %25.2lf\n",
+                       i,
+                       nrn_get_mechname(i),
+                       total_mech_count[i],
+                       static_cast<double>(total_mech_size[i]) / 1024);
```

Comment:
Might be helpful to also print out the list of mechanisms with count==0 ? i.e. debugging

Reply:
I was thinking that we already provide a list of
```
Additional mechanisms from files
 exp2syn.mod expsyn.mod halfgap.mod hh.mod netstim.mod passive.mod pattern.mod stim.mod svclmp.mod
 ```
 so there is not much added information from that